starlette/middleware/http.py

```diff
@@ -0,0 +1,103 @@
+from typing import AsyncGenerator, Callable, Optional, Union
+
+from .._compat import aclosing
+from ..datastructures import MutableHeaders
+from ..requests import HTTPConnection
+from ..responses import Response
+from ..types import ASGIApp, Message, Receive, Scope, Send
+
+_DispatchFlow = Union[
+    # Default case:
+    # response = yield
+    AsyncGenerator[None, Response],
+    # Early response and/or error handling:
+    # if condition:
+    #     yield Response(...)
+    #     return
+    # try:
+    #     response = yield None
+    # except Exception:
+    #     yield Response(...)
+    # else:
+    #    ...
+    AsyncGenerator[Optional[Response], Response],
+]
+
+
+class HTTPMiddleware:
+    def __init__(
+        self,
+        app: ASGIApp,
+        dispatch: Optional[Callable[[HTTPConnection], _DispatchFlow]] = None,
+    ) -> None:
+        if dispatch is None:
+            dispatch = self.dispatch
+
+        self.app = app
+        self._dispatch_func = dispatch
+
+    def dispatch(self, conn: HTTPConnection) -> _DispatchFlow:
```

Comment:
```suggestion
    def dispatch(self, __conn: HTTPConnection) -> _DispatchFlow:
```

This way it's a positional-only parameter in the base class and subclasses can name it as they please. Especially helpful if we end up swapping out `HTTPConnection` for some `Request` subclass in the future (`conn: Request` doesn't look very nice).

Reply:
Is that `__thing` trick an official Python thing? I see my code editor does interpret this as "positional only", as it now shows `dispatch(self, __conn: HTTPConnection, /)` on hover despite not putting the `/` separator in code.

Reply:
Yep: https://peps.python.org/pep-0484/#positional-only-arguments